service/internal/object/v1alpha2cluster/accessor.go

```diff
@@ -0,0 +1,98 @@
+package v1alpha2cluster
```

Comment:
This is nearly identical with the `machinedeployment` one. I didn't want to combine these just yet though because my thinking here is that on wider use these accessors might actually diverge quite a bit in terms of where different accessor details are eventually stored.

For example: With current AWS Cluster API implementation the API endpoint and cluster domain is not stored in `Cluster.Status` but in `AWSCluster.Spec` while in next version of Cluster API (`v1alpha3`) the API endpoint is directly in `Cluster.Spec`.